libs/librepcb/common/units/length.cpp

```diff
@@ -175,11 +175,43 @@ void Length::setLengthFromFloat(qreal nanometers) {
 
 LengthBase_t Length::mapNmToGrid(LengthBase_t  nanometers,
                                  const Length& gridInterval) noexcept {
-  if (gridInterval.mNanometers != 0)
-    return qRound((qreal)nanometers / gridInterval.mNanometers) *
-           gridInterval.mNanometers;
-  else
+  using LengthBaseU_t = std::make_unsigned<LengthBase_t>::type;
+
+  // assume gridInterval >= 0
```

Comment:
Could you make `gridInterval` absolute instead of assuming it (the old implementation also worked with negative grid interval)?

You could use `Length::toAbs()` to get the absolute value. But I have to admit that this method doesn't handle underflow properly :grin: You could either fix the handling of underflow (just map the lowest possible number to the largest possible number, although this would give an error of 1 nanometer), or just ignore that underflow (it will probably never happen anyway).

The value returned by `Length::toAbs()` could then be directly `static_cast<>`ed to an unsigned integer without worrying about underflow.

Reply:
Done. I fixed the handling of overflow in `makeAbs`.